src/protego.py

```diff
@@ -221,11 +226,20 @@ def can_fetch(self, url):
         """Return if the url can be fetched."""
         url = self._quote_path(url)
         allowed = True
+
+        # Implements selecting the applied rule by using the longest matched pattern
+        matches = []
```

Comment:
My main concern is how performance affected by this change. Before we were applying rules until first match. After everyone is applied. @anubhavp28 is it possible to reproduce [performance check ](https://anubhavp28.github.io/gsoc-weekly-checkin-12/)

Reply:
Has anyone looked at the technicalities of having a catch for wildcards? That seems to weaken the strength of the algorithm--in a semantic non-performance way.

Reply:
> Has anyone looked at the technicalities of having a catch for wildcards? That seems to weaken the strength of the algorithm--in a semantic non-performance way.

Nevermind, it's a pass not a block. I was thinking it negated that for some reason. Sorry about the noise.